packages/synapse-client/out/modelGeneric.mustache

```diff
@@ -0,0 +1,174 @@
+import { mapValues } from '../runtime{{importFileExtension}}';
+{{#hasImports}}
+{{#imports}}
+import type { {{{.}}} } from './{{.}}{{importFileExtension}}';
+import {
+    {{.}}FromJSON,
+    {{.}}FromJSONTyped,
+    {{.}}ToJSON,
+} from './{{.}}{{importFileExtension}}';
+{{/imports}}
+
+{{/hasImports}}
+{{#discriminator}}
+import {
+{{#discriminator.mappedModels}}
+     {{modelName}}FromJSONTyped{{^-last}},{{/-last}}
+{{/discriminator.mappedModels}}
+} from './index{{importFileExtension}}';
+
+{{/discriminator}}
+{{>modelGenericInterfaces}}
+
+/**
+ * Check if a given object implements the {{classname}} interface.
+ */
+{{! START CUSTOM TEMPLATING FOR synapse-client }}
+{{! Change the return type from `boolean` to a type predicate so this method can be used for type narrowing }}
+export function instanceOf{{classname}}(value: object): value is {{classname}} {
+{{! END CUSTOM TEMPLATING FOR synapse-client }}
+    {{#vars}}
+    {{#required}}
+    if (!('{{name}}' in value)) return false;
+    {{! START CUSTOM TEMPLATING FOR synapse-client }}
+    {{! if the property is a required enum, check that the value is a valid enum value }}
+    {{#isEnum}}
+    if (!Object.values({{classname}}{{enumName}}).includes(value["{{name}}"] as {{classname}}{{enumName}})) {
+      return false
+    }
+    {{/isEnum}}
+    {{! END CUSTOM TEMPLATING FOR synapse-client }}
```

Comment:
In the `instanceOf` method, if a model's property is a required enum, but the property's value does not match a known enum value, return `false`. This is especially useful because it will use our `concreteType` property, which is now always defined as an enum following PLFM-8257